### PR TITLE
Update rubocop and enable new cops

### PR DESCRIPTION
**Pull request by stephenbinns**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ inherit_gem:
 
 AllCops:
   TargetRubyVersion: 3.0
```
Comment (Tabby) [inline .rubocop.yml:7]: This should be set to the lowest version of Ruby we support with this gem, i.e. 2.7
(See https://docs.rubocop.org/rubocop/configuration.html#setting-the-target-ruby-version)
```suggestion
  TargetRubyVersion: 2.7
```

File: Gemfile [hunk 2]
```diff
@@ -11,5 +11,6 @@ elsif ENV['RAILS_VERSION']
 end
 group :development do
   # test/unit is no longer bundled with Ruby 2.2, but required by Rails
+  gem "pry"
```
Comment (Tabby) [inline Gemfile:14]: Probably ought to be `require: false` so it's not loaded when not needed
```suggestion
  gem "pry", require: false
```

**Review by Tabby — COMMENTED** [review]
Looks good. Would it be worth adding in the suggested Rubocop extensions in this PR too (rubocop-rake and rubocop-rails), or would that be better in a separate one?

File: .rubocop.yml (hunk 1 above)
Comment (stephenbinns) [reply .rubocop.yml:7]: > Looks good. Would it be worth adding in the suggested Rubocop extensions in this PR too (rubocop-rake and rubocop-rails), or would that be better in a separate one?

Probably best to split out. not sure they add much value to this if im honest

**Review by Tabby — APPROVED** [review]

**Pull request merged**